app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -1265,22 +1265,18 @@ public void setSession(@NonNull Session aSession, @OldSessionDisplayAction int a
             mViewModel.setIsDesktopMode(mSession.getUaMode() == WSessionSettings.USER_AGENT_MODE_DESKTOP);
 
             if (hidePanel) {
-                if (oldSession != null) {
-                    onCurrentSessionChange(oldSession.getWSession(), aSession.getWSession());
-                } else {
-                    onCurrentSessionChange(null, aSession.getWSession());
-                }
+                hideNewTabPanel(true);
+                hideLibraryPanel(true);
+                onCurrentSessionChange((oldSession != null ? oldSession.getWSession() : null), aSession.getWSession());
             }
 
             for (WindowListener listener: mListeners) {
                 listener.onSessionChanged(oldSession, aSession);
             }
-        }
-        mCaptureOnPageStop = false;
-
-        if (hidePanel) {
+        } else if (hidePanel) {
             closeLibrary();
         }
+        mCaptureOnPageStop = false;
     }
```

Comment:
Why we need to do this now ?

Reply:
It is in the original code, I just moved it outside of the `if`

Reply:
Understood